src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c

```diff
@@ -411,16 +416,17 @@ xocl_read_axlf_helper(struct xocl_drm *drm_p, struct drm_xocl_axlf *axlf_ptr)
 			(void) xocl_pci_rbar_refresh(xdev->core.pdev,
 					xdev->p2p_bar_idx);
 		}
+	} else if (!preserve_mem) {
+		xdev->mem_topo = vmalloc(size);
```

Comment:
393 we freed mem_topo for versal, so we malloc it at 420, what about non-versal case? should we still alloc mem here?